akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/ActorSystemStub.scala

```diff
@@ -125,3 +124,14 @@ import akka.annotation.InternalApi
 
   def address: Address = rootPath.address
 }
+
+@InternalApi private[akka] object ActorSystemStub {
+  object config {
+    // this is backward compatible with the old behaviour, hence it uses the loader used to load the test-kit
+    // which is not necessarily the one used to load the tests...
+    // hence this might not include reference config related to the actually executing test
```

Comment:
I'm not too familiar with sbt's internal but I can definitely c how it constructs its class loaders in a hierarchical manner: one level for dependencies, another for application classes, then test dependencies and finally the test classes

Reply:
Since this is a part of the testkit, it will be loaded in the "test" scope classloader together with the tests themselves as far as I can see. No class outside of the tests will trigger a load of this class/object.

Reply:
I'm also not sure I understand why we would need this defaultReference via the class loader of this class (object), but I see that was the way it was loaded previously in `ActorSystemStub`.